packages/wrangler/src/__tests__/secret.test.ts

```diff
@@ -11,6 +13,17 @@ import { useMockStdin } from "./helpers/mock-stdin";
 import { msw } from "./helpers/msw";
 import { runInTempDir } from "./helpers/run-in-tmp";
 import { runWrangler } from "./helpers/run-wrangler";
+import writeWranglerToml from "./helpers/write-wrangler-toml";
+
+jest.mock("node:readline", () => {
+	return {
+		createInterface: () =>
+			JSON.stringify({
+				secret1: "secret-value",
+				password: "hunter2",
+			}),
+	};
```

Comment:
nit? it would be better to return an AsyncIterator

```suggestion
	return {
		async function* createInterface() {
			yield JSON.stringify({
				secret1: "secret-value",
				password: "hunter2",
			}),
		},
	};
```

Reply:
Huh neat idea for making it more similar

Reply:
It turns out that a string is already an async iterator that returns each character one by one.
So given that this is a test, I didn't think it was worth the complexity.